lib/util/splice.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Plain old regular Array#splice may cause stackoverflow if we have too many items
+ */
+function chunkedSplice(array, start, deleteCount, items) {
+  var chunkSize = 10000
+  var splice = [].splice
+
+  // Make deleteCount >= 0
+  deleteCount = Math.max(deleteCount, 0)
+  // Make start between 0 and array.length (included)
+  start = Math.min(start, array.length)
+  if (start < 0) {
+    start = Math.max(array.length + start, 0)
+  }
+
+  if (items.length < chunkSize) {
+    // No need to chunk the items in this case
+    return splice.apply(array, [start, deleteCount].concat(items))
+  }
+
+  // Delete deleteCount items starting from `start`
+  var result = splice.apply(array, [start, deleteCount])
```

Comment:
Can you move the variable declarations to the top of the function?

Reply:
Also: would it perhaps be better for performance to directly copy `start..deleteCount` in from `items` (assuming `deleteCount < items.length`)?